From 808afe9e41d3d79ec875cd0cc83c84846776a4f9 Mon Sep 17 00:00:00 2001
From: Adam Williamson <awilliam@redhat.com>
Date: Thu, 7 Sep 2017 23:16:57 -0700
Subject: [PATCH] Re-introduce dnf-automatic.{service,timer}

Per RHBZ #1489595, the way the function-specific timers were
introduced has some issues. It is unnecessarily incompatible
with the previous behaviour, so systems updated from DNF 1.x
suddenly lose dnf-automatic functionality. Also, the behaviour
of the new 'function-specific' services can still in fact be
modified by the configuration file: the 'notifyonly' service
will actually download and install updates if the configuration
file says so.

This fixes both problems. It re-introduces the dnf-automatic
service and timer units with the same behaviour as before. It
also tweaks the CLI arguments and definitions of the function-
specific timers such that they will always do what they claim
to do, regardless of what the configuration file says. Finally,
it updates the dnf-automatic documentation to correctly describe
all behaviours.

Signed-off-by: Adam Williamson <awilliam@redhat.com>
---
 etc/systemd/dnf-automatic.service            | 12 ++++++++++++
 etc/systemd/dnf-automatic.timer              | 11 +++++++++++
 10 files changed, 73 insertions(+), 21 deletions(-)
 create mode 100644 etc/systemd/dnf-automatic.service
 create mode 100644 etc/systemd/dnf-automatic.timer

diff --git a/etc/systemd/dnf-automatic.service b/etc/systemd/dnf-automatic.service
new file mode 100644
index 00000000..0b390c8a
--- /dev/null
+++ b/etc/systemd/dnf-automatic.service
@@ -0,0 +1,12 @@
+[Unit]
+Description=dnf automatic
+# See comment in dnf-makecache.service
+ConditionPathExists=!/run/ostree-booted
+
+[Service]
+Type=oneshot
+Nice=19
+IOSchedulingClass=2
+IOSchedulingPriority=7
+Environment="ABRT_IGNORE_PYTHON=1"
+ExecStart=/usr/bin/dnf-automatic /etc/dnf/automatic.conf --timer
diff --git a/etc/systemd/dnf-automatic.timer b/etc/systemd/dnf-automatic.timer
new file mode 100644
index 00000000..62f1b70d
--- /dev/null
+++ b/etc/systemd/dnf-automatic.timer
@@ -0,0 +1,11 @@
+[Unit]
+Description=dnf-automatic timer
+# See comment in dnf-makecache.service
+ConditionPathExists=!/run/ostree-booted
+
+[Timer]
+OnBootSec=1h
+OnUnitInactiveSec=1d
+
+[Install]
+WantedBy=basic.target
-- 
2.13.5

